BREAKING.md

```diff
@@ -1,3 +1,9 @@
+## 0.39 Breaking changes
+- [connect event removed from Container](#connect-event-removed-from-Container)
+
+### connect event removed from Container
+The `"connect"` event would previously fire on the `Container` after `connect_document_success` was received from the server (which likely happens before the client's own join message is processed).  This event does not represent a safe-to-use state, and has been removed.  To detect when the `Container` is fully connected, the `"connected"` event should be used instead.
```

Comment:
I think this is moving in the wrong direction. First of all, "connected" does not provide that guarantee (consider "read" connection"). But most important, I believe hosts should not care, what they care are two distinct signals:
- Is there a connection (to show various warning / notifications to users when there is none)
- is it safe to close container and not lose user state ("dirty" flag).
And combination of them to have proper message ("your changes are not saved because there is no connection", "your changes are not saved, but we are working on it..." (aka "saving..."), etc.).

Attempts to pivot on connected would always be not sufficient, because you need to consider if all changes were acked by server (i.e. container dirty or not dirty).

I believe right thing to do here is
- remove connected event
- expose better tools for scenarios that our users need, like get self "IClient" and get notifications when it changes (i.e. be able to find your self in audience / quorum, and as much as possible hide at that level notion of clientID).

The picture is different for "inside" container, as there needs are different - one needs to know when it safe to send ops, and track clientIDs (mostly DDSs).

This is all from POV of separating host from component / container authors. In a world where it's one and the same, as should push them to be "inside" container.